ziti/client.go

```diff
@@ -0,0 +1,368 @@
+package ziti
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/edge-api/rest_client_api_client"
+	"github.com/openziti/edge-api/rest_client_api_client/authentication"
+	"github.com/openziti/edge-api/rest_client_api_client/current_api_session"
+	"github.com/openziti/edge-api/rest_client_api_client/current_identity"
+	"github.com/openziti/edge-api/rest_client_api_client/posture_checks"
+	"github.com/openziti/edge-api/rest_client_api_client/service"
+	"github.com/openziti/edge-api/rest_client_api_client/session"
+	"github.com/openziti/edge-api/rest_model"
+	"github.com/openziti/edge-api/rest_util"
+	nfPem "github.com/openziti/foundation/v2/pem"
+	"github.com/openziti/identity"
+	"github.com/openziti/sdk-golang/ziti/edge/posture"
+	"net/url"
+	"time"
+)
+
+// CtrlClient is a stateful version of ZitiEdgeClient that simplifies operations
+type CtrlClient struct {
+	*rest_client_api_client.ZitiEdgeClient
+	Authenticator rest_util.Authenticator
+
+	ApiSession *rest_model.CurrentAPISessionDetail
+
+	lastServiceUpdate  *strfmt.DateTime
+	lastServiceRefresh *strfmt.DateTime
+
+	EdgeClientApiUrl *url.URL
+
+	ApiSessionIdentity          identity.Identity
+	ApiSessionCertificateDetail rest_model.CurrentAPISessionCertificateDetail
+	ApiSessionCsr               x509.CertificateRequest
+	ApiSessionCertificate       *x509.Certificate
+	ApiSessionPrivateKey        *ecdsa.PrivateKey
+	CaPool                      *x509.CertPool
+	ApiSessionCertInstance      string
+
+	PostureCache *posture.Cache
+}
+
+func (client *CtrlClient) AuthenticateRequest(request runtime.ClientRequest, registry strfmt.Registry) error {
```

Comment:
this whole file should get godoc'ed, right?

Reply:
Mmm, I was going to say that the edge-API doc covers it, but I can see a reason to add some light documentation here.